packages/toolpad-app/src/toolpad/AppEditor/AppAuthorizationEditor.tsx

```diff
@@ -693,22 +696,37 @@ export default function AppAuthorizationDialog({ open, onClose }: AppAuthorizati
             <TabPanel disableGutters value="authentication">
               <AppAuthenticationEditor />
             </TabPanel>
-            <TabPanel disableGutters value="roles">
-              <Typography variant="body2">
-                Define the roles for your application. You can configure your pages to be accessible
-                to specific roles only.
-              </Typography>
-              <AppRolesEditor onRowUpdateError={handleRowUpdateError} />
-            </TabPanel>
-            <TabPanel disableGutters value="roleMappings">
-              <Typography variant="body2">
-                Define mappings from authentication provider roles to Toolpad roles.
-              </Typography>
-              <AppRoleMappingsEditor
-                onRowUpdateError={handleRowUpdateError}
-                roleEnabledActiveAuthProviderOptions={roleEnabledActiveAuthProviderOptions}
-              />
-            </TabPanel>
+
+            <React.Fragment>
+              <TabPanel disableGutters value="roles">
+                {isPaidPlan ? (
+                  <React.Fragment>
+                    <Typography variant="body2">
+                      Define the roles for your application. You can configure your pages to be
+                      accessible to specific roles only.
+                    </Typography>
+                    <AppRolesEditor onRowUpdateError={handleRowUpdateError} />
+                  </React.Fragment>
+                ) : (
+                  <ErrorUpgrade feature="Roles" />
+                )}
+              </TabPanel>
+              <TabPanel disableGutters value="roleMappings">
+                {isPaidPlan ? (
```

Comment:
The role mappings tab only shows if at least one authentication provider that supports roles is selected. If we end up not letting users use even just authentication in any provider that supports roles, we wouldn't need the check in the role mappings tab. Anyway it's not very important probably, just mentioning it!

Reply:
Makes sense! I've added a disabling paid authentication types from the Authentication selection section, but I feel we can keep as a way to keep the knowledge that role mappings is a paid feature in the codebase.